build/ci/phase-template.yml

```diff
@@ -36,5 +38,23 @@ phases:
         testRunTitle: Machinelearning_Tests_$(_phaseName)_$(_configuration)_$(Build.BuildNumber)
         configuration: $(_configuration)
         mergeTestResults: true
+    - task: CopyFiles@2
+      displayName: Stage build logs
+      inputs:
+        sourceFolder: $(Build.SourcesDirectory)
+        contents: '?(msbuild.*|binclash.log|init-tools.log)'
+        targetFolder: $(Build.ArtifactStagingDirectory)
```

Comment:
I think you need to set a condition to run this always even if a previous step failed. This to guarantee always getting logs. The same way as publish test results step.

Reply:
Good point.  I think I'll specify "failed" only because that is when we care about the logs.